### PR TITLE
add azure UltraSSD, StandardSSD disk type support

### LINKED ISSUE #70220: add azure UltraSSD disk support
**What would you like to be added**:
https://docs.microsoft.com/en-us/azure/virtual-machines/linux/disks-ultra-ssd

**Why is this needed**:
Need to upgrade SDK support to `2018-10-01/compute` (current is `2018-04-01/compute`)

/kind feature

/assign
/sig azure

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
add azure UltraSSD disk support

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #70220

**Special notes for your reviewer**:
In this PR,  added two new parameters in azure disk storage class: `diskIopsReadWrite` (default: 1200), `diskMbpsReadWrite` (default: 300) according to https://docs.microsoft.com/en-us/azure/virtual-machines/linux/disks-ultra-ssd.
You may review the second commit for mainly related code change:https://github.com/kubernetes/kubernetes/pull/70477/commits/f1bd292c764e5dac507c5d5ca52784b5d3c04c4d
The first commit is go sdk upgrade.

```
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: ultrassd
provisioner: kubernetes.io/azure-disk
parameters:
  skuname: UltraSSD_LRS
  kind: managed
  cachingmode: None
  diskIopsReadWrite: "1400"
  diskMbpsReadWrite: "320"
```
btw, possible values of `skuname` could be: `Standard_LRS`, `StandardSSD_LRS`, `Premium_LRS`, `UltraSSD_LRS`

**Does this PR introduce a user-facing change?**:

```
NONE
```

**Release note**:
```
add azure UltraSSD disk support
```
/sig azure

/assign @khenidak @feiskyer 
cc @brendandburns

### CODE REVIEW

**Comment by andyzhangx** [comment]
FYI @sylus

File: pkg/cloudprovider/providers/azure/azure_managedDiskController.go [hunk 1]
```diff
@@ -87,17 +97,42 @@ func (c *ManagedDiskController) CreateManagedDisk(options *ManagedDiskOptions) (
...
+		DiskSizeGB:   &diskSizeGB,
+		CreationData: &compute.CreationData{CreateOption: compute.Empty},
+	}
+
+	if diskSku == compute.UltraSSDLRS {
+		diskIOPSReadWrite := int64(defaultDiskIOPSReadWrite)
+		if options.DiskIOPSReadWrite != "" {
+			v, err := strconv.Atoi(options.DiskIOPSReadWrite)
+			if err != nil {
+				return "", err
```
Comment (feiskyer) [inline pkg/cloudprovider/providers/azure/azure_managedDiskController.go]: nit: Add "DiskIOPSReadWrite" to error message

File: pkg/cloudprovider/providers/azure/azure_managedDiskController.go [hunk 2]
```diff
@@ -87,17 +97,42 @@ func (c *ManagedDiskController) CreateManagedDisk(options *ManagedDiskOptions) (
...
+			}
+			diskIOPSReadWrite = int64(v)
+		}
+		diskProperties.DiskIOPSReadWrite = to.Int64Ptr(diskIOPSReadWrite)
+
+		diskMBpsReadWrite := int32(defaultDiskMBpsReadWrite)
+		if options.DiskMBpsReadWrite != "" {
+			v, err := strconv.Atoi(options.DiskMBpsReadWrite)
+			if err != nil {
+				return "", err
```
Comment (feiskyer) [inline pkg/cloudprovider/providers/azure/azure_managedDiskController.go]: nit: Add "DiskMBpsReadWrite" to error message

File: pkg/volume/azure_dd/azure_provision.go [hunk 3]
```diff
@@ -241,6 +249,8 @@ func (p *azureDiskProvisioner) Provision(selectedNode *v1.Node, allowedTopologie
 			SizeGB:             requestGiB,
 			Tags:               tags,
 			AvailabilityZone:   selectedAvailabilityZone,
+			DiskIOPSReadWrite:  diskIopsReadWrite,
+			DiskMBpsReadWrite:  diskMbpsReadWrite,
```
Comment (feiskyer) [inline pkg/volume/azure_dd/azure_provision.go]: Those two options only work for UltraSSD disk. Add a pre-check here and fail out if the disk is not UltraSSD.

File: pkg/cloudprovider/providers/azure/azure_managedDiskController.go (hunk 1 above)
Comment (andyzhangx) [reply pkg/cloudprovider/providers/azure/azure_managedDiskController.go]: fixed

File: pkg/cloudprovider/providers/azure/azure_managedDiskController.go (hunk 2 above)
Comment (andyzhangx) [reply pkg/cloudprovider/providers/azure/azure_managedDiskController.go]: fixed

File: pkg/volume/azure_dd/azure_provision.go (hunk 3 above)
Comment (andyzhangx) [reply pkg/volume/azure_dd/azure_provision.go]: fixed

**Comment by andyzhangx** [comment]
@brendandburns @khenidak PTAL
the k8s upstream support for UltraSSD is done with this PR, while UltraSSD feature only supports availability zone without vmss, while our acs-engine could only create availability zone with vmss, so we are looking forward to UltraSSD support on vmss.

**Comment by mbohlool** [comment]
cc @lavalamp

File: staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/azure_test.go [hunk 4]
```diff
@@ -125,8 +126,8 @@ func newFackeAzureToken(accessToken string, expiresOn string) adal.Token {
 		AccessToken:  accessToken,
 		RefreshToken: "fake",
 		ExpiresIn:    "3600",
-		ExpiresOn:    expiresOn,
-		NotBefore:    expiresOn,
+		ExpiresOn:    json.Number(expiresOn),
+		NotBefore:    json.Number(expiresOn),
```
Comment (lavalamp) [inline staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/azure_test.go]: Will old clients work with new servers and vice versa?

**Comment by lavalamp** [comment]
/approve

I can't tell if there's a compatibility problem or not, I'll add a hold to make sure it doesn't merge until you see this; do a "/hold cancel" if there isn't a problem.

/hold

File: staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/azure_test.go (hunk 4 above)
Comment (andyzhangx) [reply staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/azure_test.go]: yes, `json.Number` is actually string:
```
// A Number represents a JSON number literal.
type Number string
```
https://github.com/golang/go/blob/914a25c0eb8aacdbb5cddd3e7d5e6ee7b019f88f/src/encoding/json/decode.go#L187

**Comment by feiskyer** [comment]
@andyzhangx Seems release notes format is wrong? Could you add a release note for this?

**Comment by andyzhangx** [comment]
@feiskyer done, original PR template is missing release notes item.

**Pull request merged**